src/partition/ProvidedPartition.cpp

```diff
@@ -104,7 +104,8 @@ void ProvidedPartition::communicate()
       Event e("partition.sendGlobalMesh." + _mesh->getName(), precice::syncMode);
 
       if (not utils::MasterSlave::isSlave()) {
-        PRECICE_CHECK(globalMesh.vertices().size() > 0, "The provided mesh " << globalMesh.getName() << " is invalid (possibly empty).");
+        PRECICE_CHECK(globalMesh.vertices().size() > 0,
+                      "The provided mesh \"" << globalMesh.getName() << "\" was not defined. Please set the mesh using setMeshXXX() prior to calling initialize().");
```

Comment:
```suggestion
                      "The provided mesh \"" << globalMesh.getName() << "\" is empty. Please set the mesh using setMeshXXX() prior to calling initialize().");
```
"defined" is a bit ambiguous. We could say that one defines a mesh already in the config.

Reply:
I am fine with the change.

Normally declaration means "there is a mesh" and definition means "this is how the mesh looks".
But my sight is probably tainted by the programmer inside me.